include/boost/spirit/home/qi/directive/expect.hpp

```diff
@@ -0,0 +1,115 @@
+/*=============================================================================
+Copyright (c) 2016 Frank Hein, maxence business consulting gmbh
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+==============================================================================*/
+#if !defined (SPIRIT_EXPECT_JULY_13_2016)
+#define SPIRIT_EXPECT_JULY_13_2016
+
+#if defined(_MSC_VER)
+#pragma once
+#endif
+
+#include <boost/spirit/home/qi/meta_compiler.hpp>
+#include <boost/spirit/home/support/has_semantic_action.hpp>
+#include <boost/spirit/home/qi/detail/attributes.hpp>
+#include <boost/spirit/home/support/common_terminals.hpp>
+#include <boost/spirit/home/support/handles_container.hpp>
+#include <boost/spirit/home/support/unused.hpp>
+#include <boost/spirit/home/support/info.hpp>
+
+#include <stdexcept>
+
+namespace boost { namespace spirit {
+    ///////////////////////////////////////////////////////////////////////////
+    // Enablers
+    ///////////////////////////////////////////////////////////////////////////
+    template <>
+    struct use_directive<qi::domain, tag::expect> // enables expect[p]
+        : mpl::true_ {};
+}}
+
+namespace boost { namespace spirit { namespace qi {
+
+#ifndef BOOST_SPIRIT_NO_PREDEFINED_TERMINALS
+    using spirit::expect;
+#endif
+    using spirit::expect_type;
+
+    template <typename Subject>
+    struct expect_directive : unary_parser<expect_directive<Subject> >
+    {
+        using subject_type 
+            = result_of::compile<domain, Subject>;
+
```

Comment:
Ooops. Will this work with C++03? Have we checked if the additions work with C++03? That's our minimum for Qi and Karma (not so for X3).